Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,259 @@
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the function to destroy the connection and clean the variables related
+     *
+     * @dataProvider getMessages
+     * @group destroy
+     */
+    public function testDestroy(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $this->driver->declareChannel();
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->consume($this->consumer->getName(), $this->queueName);
+        $this->driver->destroy($this->consumer->getName());
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+     /**
+     * Tests that tries to declare a channel without having a connection
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
+    }
+
+    /**
+     * Test the format parameter set in the class
+     */
+    public function testFormat()
+    {
+        $format = 'text/plain';
+        $this->driver->setFormat($format);
+        $this->assertEquals($format, $this->driver->getFormat());
+    }
+
+    /**
+     * @throws AMQPProtocolException
+     */
+    public function testConnectWithoutData()
+    {
+        $this->driver->declareChannel();
+        $consumer = $this->createConsumer();
+        $this->driver->destroy($consumer->getName());
+        $this->driver->configure('', '', '', '');
+        $this->driver->connect();
+        $this->assertTrue($this->driver->isConnected());
+        $this->assertInstanceOf(AMQPStreamConnection::class, $this->driver->declareChannel()->getConnection());
+    }
+
+    /**
+     * @dataProvider getMessages
+     */
+    public function testConsumeWaitNoBlock(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage){
+            $amqpMessage = $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($message->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->waitNoBlock();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
```

Comment:
Is there an assert way we can validate that the waitNoBlock is working as expected?

Reply:
I don't know how to assert this, otherwise calling between the process and expecting the consumer response. The main difference is the time that it takes to process once this ignore the timeout parameter.

Reply:
@andreskrey do you have any idea on how to do this?

Reply:
Passing a callback and performing assertions with the received $amqpMessage validates the correct functioning of `waitNoBlock()`. In my opionion!

Reply:
Ok